Verify previous Terraform state can be reused

NamelessOne91 opened the pull request:
Related https://github.com/SUSE/spacewalk/issues/25242

Introduces additional checks when "use_previous_terraform_state"is checked,  with the aim to validate a previous Terraform state is indeed present and can be safely reused.

NamelessOne91 committed:
Verify previous Terraform state can be reused

srbarrios commented:
Looks a nice improvement.
As suggestion, I would also try to use virsh to collect the actual resources status (for instance VMs and disks created), as the tfstate it might actually don't match the real state.
And if I'm guessing right, terraform plan only compares the information from tfstate and main.tf, without directly checking the real deployed resources.

NamelessOne91 commented:
> And if I'm guessing right, terraform plan only compares the information from tfstate and main.tf, without directly checking the real deployed resources.

IIRC plan does automatically a refresh to check differences between deployed resources and what's in the state file ... but I may not RC 😄 
So, it seems there's an option to force it anyway.

I added it and refactored the code a bit to be more robust. 
Still needs to be tested properly, as I have a few doubts it will work and do what we expect.
